File: Solutions/2210. Count Hills and Valleys in an Array/2210. Count Hills and Valleys in an Array.go
func countHillValley(nums []int) int {
    var tmp []int
    for i,_:=range nums{
        if i==0 || nums[i]!=nums[i-1]{
            tmp=append(tmp,nums[i])
        }
    }
    res:=0
    for i:=1;i<len(tmp)-1;i+=1{
        if (tmp[i-1]<tmp[i] && tmp[i]>tmp[i+1]) || (tmp[i-1]>tmp[i] && tmp[i]<tmp[i+1]){
            res+=1
        }
    }
    return res
}

File: Solutions/2210. Count Hills and Valleys in an Array/2210. Count Hills and Valleys in an Array.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums) {
        vector<int> tmp;
        for(int i=0;i<nums.size();++i){
            if(i==0 or nums[i]!=nums[i-1]){
                tmp.push_back(nums[i]);
            }
        }
        int res=0;
        for(int i=1;i<tmp.size()-1;++i){
            if((tmp[i-1]<tmp[i] and tmp[i]>tmp[i+1]) or (tmp[i-1]>tmp[i] and tmp[i]<tmp[i+1])){
                ++res;
            }
        }
        return res;
    }
};